### PR TITLE
Some GXM implementations and a few other fixes

### PR DESCRIPTION
**Pull request by VelocityRa**

### CODE REVIEW

File: src/emulator/host/src/app.cpp [hunk 1]
```diff
@@ -309,7 +309,8 @@ ExitCode load_app(Ptr<const void> &entry_point, HostState &host, const std::wstr
         auto &imgui_render = host.display.imgui_render;
 
         bool old_imgui_render = imgui_render.load();
-        while (!imgui_render.compare_exchange_weak(old_imgui_render, !old_imgui_render));
+        while (!imgui_render.compare_exchange_weak(old_imgui_render, !old_imgui_render))
+            ;
```
Comment (Zangetsu38) [inline src/emulator/host/src/app.cpp]: Sorry, for here, and i have see, i have set one think no needed here you can remove While, with set just like that so 
>!imgui_render.compare_exchange_weak(old_imgui_render, !old_imgui_render);

Comment (pent0) [reply src/emulator/host/src/app.cpp]: The while was intentional, it was supposed to lock the current thread until the conditional is satisfied.

Comment (Zangetsu38) [reply src/emulator/host/src/app.cpp]: Oh i see i have think is needed only for loop 
 With press g

File: src/emulator/modules/SceGxm/SceGxm.cpp [hunk 2]
```diff
@@ -858,56 +858,61 @@ EXPORT(int, sceGxmProgramIsSpriteCoordUsed) {
...
+    uint64_t parameter_offset = program->parameters_offset;
+
+    if (parameter_offset > 0)
+        parameter_offset += (uint64_t)&program->parameters_offset;
+    return (uint32_t)((uint64_t)parameter - parameter_offset) >> 4;
 }
 
-EXPORT(Ptr<void>, sceGxmProgramParameterGetName, Ptr<const SceGxmProgramParameter> parameter) {
-    return Ptr<void>(parameter.address() + parameter.get(host.mem)->name_offset);
+EXPORT(Ptr<const char>, sceGxmProgramParameterGetName, Ptr<const SceGxmProgramParameter> parameter) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceGxm/SceGxm.cpp]: @petmac Can this be rewritten to use raw C pointers? I'm not sure what the recent (?) bridging change was about.

Comment (petmac) [reply src/emulator/modules/SceGxm/SceGxm.cpp]: I don't think so, because there's no safe way to convert back from a raw pointer to an address (Ptr).

Comment (VelocityRa) [reply src/emulator/modules/SceGxm/SceGxm.cpp]: I see, thanks.

**Pull request merged**